net/basic_socket.cpp

```diff
@@ -262,11 +262,11 @@ bool ISocketStream::skip_read(size_t count) {
 }
 
 int do_get_name(int fd, Getter getter, EndPoint& addr) {
-    struct sockaddr_in addr_in;
-    socklen_t len = sizeof(addr_in);
-    int ret = getter(fd, (struct sockaddr*) &addr_in, &len);
-    if (ret < 0 || len != sizeof(addr_in)) return -1;
-    addr.from_sockaddr_in(addr_in);
+    sockaddr_storage storage(addr);
```

Comment:
Is it possible that ```addr``` is not initialized?

Reply:
Added a new `get_max_socklen`  to `sockaddr_storage`.